core/message_pool.go

```diff
@@ -209,3 +246,42 @@ func (pool *MessagePool) LargestNonce(address address.Address) (largest uint64,
 	}
 	return
 }
+
+// ValidateMessage in message pool is a mechanism for preventing memory based DoS attacks.
+// As such, it will often fail silently.
+func (pool *MessagePool) ValidateMessage(ctx context.Context, message *types.SignedMessage) error {
+	if len(pool.pending) >= MaxMessagePoolSize {
+		return ErrMessagePoolFull
+	}
+
+	// check that message with this nonce does not already exist
+	_, found := pool.addressNonces[newAddressNonce(message)]
+	if found {
+		return ErrDuplicateNonce
+	}
+
+	st, err := pool.api.LatestState(ctx)
+	if err != nil {
+		return err
+	}
+
+	fromActor, err := st.GetActor(ctx, message.From)
+	if err != nil {
+		if state.IsActorNotFoundError(err) {
+			fromActor = &actor.Actor{}
+		} else {
+			return err
+		}
+	}
+
+	// check that message nonce is not too high
+	if message.Nonce > fromActor.Nonce && message.Nonce-fromActor.Nonce > MaxNonceGap {
+		return ErrNonceGapExceeded
+	}
+
+	if err = consensus.NewOutboundMessageValidator().Validate(ctx, message, fromActor); err != nil {
```

Comment:
I think we can skip the err check here:
```suggestion
return consensus.NewOutboundMessageValidator().Validate(ctx, message, fromActor)
```